conbench/entities/case.py

```diff
@@ -13,4 +15,24 @@ class Case(Base, EntityMixin):
     tags: Mapped[dict] = NotNull(postgresql.JSONB)
 
 
+def get_case_or_create(case_dict) -> Case:
+    """
+    Try to create case, but expect conflict (work with unique constraint on
+    name/tags).
+
+    Return (newly created, or previously existing) Case object, or raise an
+    exception.
+    """
+    try:
+        return Case.create(case_dict)
+    except s.exc.IntegrityError as exc:
+        if "violates unique constraint" not in str(exc):
+            raise
+
+    Session.rollback()
```

Comment:
The rollback is necessary because we got here because the previous transaction failed.

In the future, we can apply an automatic rollback strategy:
https://docs.sqlalchemy.org/en/14/orm/session_basics.html#framing-out-a-begin-commit-rollback-block
https://stackoverflow.com/a/52234206/145400